Update JAX-export tutorial

pavithraes opened the pull request:
This PR updates "Exporting StableHLO from JAX" tutorial. Specifically:

- Move from `jax.experimental.export` to `jax.export`, mainly involves explicit use of `jax.jit`
- Modification of narrative text for accuracy, consistency, and additional links
- Add a couple of introductory sentences

GleasonK commented on docs/tutorials/jax-export.ipynb:
```diff
@@ -1,445 +1,479 @@
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/Users/pavithraes/mambaforge/envs/shlo-docs/lib/python3.11/pty.py:89: RuntimeWarning: os.fork() was called. os.fork() is incompatible with multithreaded code, and JAX is multithreaded, so this will likely lead to a deadlock.\n",
```
<img width="1086" alt="image" src="https://github.com/user-attachments/assets/7827045f-0421-4110-ace2-50824b8cc3bf">

Can we see if there's a way to suppress errors? Not sure if there's a way to write this code without some implicit call to fork, but don't want to show your user path here 🙂

pavithraes replied:
Oh right, I was meaning to clean this but missed it. Will do!

GleasonK reviewed (changes requested):
Looks great, thank you! One thing to fix then happy to merge.

pavithraes commented:
Thanks, I've addressed your comment. I also included a minor update to keep the structure (installation & troubleshooting) consistent with the proposed updates to the PyTorch export docs.

This is ready for final review & merge. :)
